lib/evaluation/handlers/sameSubjectEvaluationHandler.ts

```diff
@@ -30,223 +21,162 @@ export class SameSubjectEvaluationHandler extends AbstractEvaluationHandler {
-    return (descriptor: Descriptor) => this.findDescribedCredentialPaths(descriptor);
-  }
-
-  private findDescribedCredentialPaths(descriptor: Descriptor): void {
-    this.allDescribedCredentialsPaths.set(descriptor.id, descriptor.path);
-
-    if (descriptor.path_nested) {
-      this.findDescribedCredentialPaths(descriptor.path_nested);
-    }
-  }
-
-  private findAllCredentialSubjects() {
-    this.allDescribedCredentialsPaths.forEach(this.mapCredentialPathToCredentialSubject());
-  }
-
-  private mapCredentialPathToCredentialSubject() {
-    return (path: string, inDescId: string) => {
-      const subjectNode: CredentialSubjectJsonpathNode[] = jp.nodes(
-        this.vPresentation,
-        path.concat('.credentialSubject')
-      );
-      if (subjectNode.length) {
-        this.credentialsSubjects.set(inDescId, subjectNode[0].value);
+    const error: [string, string[]][] = [];
+    sameSubject
+      .filter((d) => d.value.directive === directive)
+      .filter((e) => e.value.field_id.every((id) => fields.includes(id)))
+      .forEach((p) => fieldsMapping.set(jp.stringify(p.path.slice(0, 3)), p.value.field_id));
+
+    sameSubject
+      .filter((d) => d.value.directive === directive)
+      .filter((e) => !e.value.field_id.every((id) => fields.includes(id)))
+      .forEach((p) => error.push([jp.stringify(p.path.slice(0, 3)), p.value.field_id]));
+    return error;
+  }
+
+  private findAllCredentialSubjects(vcs: VerifiableCredential[]) {
+    //TODO handle nested path
+    const credentialSubject: { path: PathComponent[]; value: CredentialSubject }[] = jp.nodes(
+      vcs,
+      '$..credentialSubject'
+    );
+    credentialSubject.forEach((cs) => this.credentialsSubjects.set(jp.stringify(cs.path.slice(0, 2)), cs.value));
+  }
+
+  private confirmAllFieldSetHasSameSubject(fieldIdzInputDescriptorsGroups: Map<string, string[]>, status: Status) {
+    const credentialMatchingFields = Array.from(fieldIdzInputDescriptorsGroups).flatMap((k) =>
+      Array.from(this.credentialsSubjects).filter((a) => k[1].find((c) => Object.keys(a[1]).includes(c)))
+    );
+
+    const fields = Array.from(credentialMatchingFields).flatMap((s) => Object.keys(s[1]).filter((w) => w !== 'id'));
+
+    const allFieldsMatched: boolean = Array.from(fieldIdzInputDescriptorsGroups.values()).flatMap((v) =>
+      v.every((e) => fields.includes(e))
+    )[0];
+
+    const isSameSubject: boolean =
+      new Set(Array.from(credentialMatchingFields).flatMap((s) => Object.keys(s[1]).filter((w) => w === 'id'))).size ===
+      1;
+
+    const credentialsToInputDescriptors = this.mapCredentialsToInputDescriptors();
+
+    credentialMatchingFields.forEach((subject) => {
+      const inDescPath: string = credentialsToInputDescriptors.get(subject[0]) as string;
+      if (allFieldsMatched && isSameSubject) {
+        this.getResults().push(this.createResult(fields, inDescPath, subject, status));
+      } else {
```

Comment:
The double negative in your comment is confusing. You are basically saying it is correct, but you probably mean the opposite

Reply:
The presentation submission is basically a mapping between input descriptors and vcs. PE-JS uses the logs generated on each evaluation handler to create the presentation submission. There was a break of contract in the previous implementation because there was no input descriptor to credential mapping and same subject was totally ignored during the presentation submission creation. The way it is done now there is a group matching and input descriptor to credential mapping, Thus I don't understand what is incorrect.

Reply:
@nklomp I have removed the double negative. It was a mistake in my comment

Reply:
Due to difference of opinion we need to write a test case for which I can provide the data. If the test case passes we can mark this resolved.